### PR TITLE
Added Powi

### PR DESCRIPTION
**Pull request by Cazadorro**
>removed un-necessary code in fract and utils, added MANTISSA_BITS which is the number of bits reserved for the f32 mantissa (23), and added powi

Was working on another project which needed this functionality in an environment where it wasn't built in.  Saw rust had this functionality as well in std and decided to just implement it for micromath

### CODE REVIEW

File: src/f32ext/powi.rs [hunk 1]
```diff
@@ -0,0 +1,2681 @@
...
+#[cfg(test)]
+mod tests {
+    use super::super::abs;
+    use super::powi_exp_by_squaring;
+    use core::f32;
+
+    //error builds up from both exp and ln approximation, so we double the error allowed.
+    pub(crate) const MAX_ERROR: f32 = 0.002;
+    ///  powi(3,x) test vectors - `(input_base, input_power, output)`
+    pub(crate) const TEST_VECTORS: &[(f32, f32, f32)] = &[
```
Comment (tarcieri) [inline src/f32ext/powi.rs:69]: Yowza

**Review by tarcieri — APPROVED** [review]

**Comment by tarcieri** [comment]
Nice, looks good!

**Comment by regexident** [comment]
Any chance to get a release cut with this addition, @tarcieri? 🙏🏻

**Comment by tarcieri** [comment]
Released in 1.1.0

**Comment by regexident** [comment]
Thanks!

**Pull request merged**